s3/s3.go

```diff
@@ -0,0 +1,243 @@
+package s3
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	awsS3 "github.com/aws/aws-sdk-go/service/s3"
+	"github.com/taskcluster/taskcluster-cli/config"
+	"github.com/taskcluster/taskcluster-cli/extpoints"
+	"github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-client-go/auth"
+)
+
+func init() {
+	extpoints.Register("s3", S3{})
+}
+
+type S3 struct{}
+
+func (S3) ConfigOptions() map[string]extpoints.ConfigOption {
+	return nil
+}
+
+func (S3) Summary() string {
+	return "Uploads/downloads file to/from AWS S3."
+}
+
+func usage() string {
+	return `Upload/Download file to/from AWS S3.
+Usage:
+  taskcluster S3 put <file> <bucket> <prefix>
+  taskcluster S3 get <key> <bucket> <prefix>
+  taskcluster S3 help <subcommand>
+`
+}
+
+func (S3) Usage() string {
+	return usage()
+}
+
+func (S3) Execute(context extpoints.Context) bool {
+	argv := context.Arguments
+
+	// Check for command
+	command := argv["S3"].(string)
+	provider := extpoints.CommandProviders()[command]
+	if provider == nil {
+		panic(fmt.Sprintf("Unknown command: %s", command))
+	}
+
+	// Print help for subcommands
+	if argv["help"] == true {
+		subcommand, ok := argv["<subcommand>"].(string)
+		if !ok {
+			panic(fmt.Sprintf("Unknown subcommand: %s", subcommand))
+		}
+		if subcommand == "put" {
+			fmt.Println("Usage:\ntaskcluster S3 put <file> <bucket> <prefix>")
+		} else if subcommand == "get" {
+			fmt.Println("Usage:\ntaskcluster S3 get <key> <bucket> <prefix>")
+		} else {
+			fmt.Printf("Invalid subcommand.\n")
+		}
+		return true
+	}
+
+	// Parse for bucket and prefix
+	bucket, ok := argv["<bucket>"].(string)
+	if !ok {
+		fmt.Println("Invalid bucket format.")
+		return false
+	}
+	prefix, ok := argv["<prefix>"].(string)
+	if !ok {
+		fmt.Println("Invalid prefix format.")
+		return false
+	}
+
+	// Set level, filename and key
+	var level, filename, key string
+	if argv["put"] == true {
+		level = "read-write"
+		filename, ok = argv["<file>"].(string)
+		if !ok {
```

Comment:
My point is that there is no reason not to do `filename = args["<file>"].(string)` because if `argv["put"] == true` then the required arguments must be present, see:
https://github.com/taskcluster/taskcluster-cli/blob/master/config/config.go#L183
There we don't do the conditional check for `filename, ok = argv["<file>"].(string)`, because the `ok` check is unnecessary.